src/main/java/com/marek/onlinebookstore/repository/book/BookRepository.java

```diff
@@ -12,12 +10,7 @@
 
 @Repository
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+
     @Query("SELECT b FROM Book b JOIN FETCH b.categories c WHERE c.id = :categoriesId")
     List<Book> findAllByCategoriesId(Long categoriesId, Pageable pageable);
```

Comment:
The parameter 'categoriesId' in the method findAllByCategoriesId should be 'categoryId' as it is supposed to take a single category id, not multiple.